service/create/service.go

```diff
@@ -1256,7 +1254,7 @@ func (s *Service) deleteFunc(obj interface{}) {
 	// Delete ingress security group.
 	ingressSGInput := securityGroupInput{
 		Clients:   clients,
-		GroupName: securityGroupName(cluster.Name, prefixIngress),
+		GroupName: securityGroupName(key.ClusterID(cluster), prefixIngress),
```

Comment:
Security group name should also be a key function and just take the cluster and the prefix then.